### PR TITLE
Feature/dataset download

### PR DESCRIPTION
**Pull request by jamie-short**
added in command `dafni download dataset {dataset_id} {version_id}`
- This downloads all files to a zip file in the current workng directory by default
- File name `Dataset_{dataset_id]_{version_id}.zip`
- a `--directory` option can be given to define a different location to the cwd

### CODE REVIEW

File: test/test_utils.py [hunk 1]
```diff
@@ -771,6 +776,32 @@ def test_arguments_are_printed_correctly_with_none_additional_message_specified(
...
-        mock_click.confirm.assert_called_once_with(confirmation_message, abort=True)
\ No newline at end of file
+        mock_click.confirm.assert_called_once_with(confirmation_message, abort=True)
+
+
+class TestWriteFilesToZip:
+    """Test class to test the write_files_to_zip functionality"""
+
+    def test_given_files_written_as_zip_file_to_given_location_with_correct_contents(
+        self, tmp_path
```
Comment (iroper-tessella) [inline test/test_utils.py:788]: I'm a little confused how this is working. There is no patch for `tmp_path`.

**Review by iroper-tessella — CHANGES_REQUESTED** [review]
A few questions (which I won't be hear to hear the answer from but anyway...) and a few things to change.

File: dafni_cli/commands/download.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    version_id: List[str],
+    directory: Optional[click.Path],
+):
+    """Download all files associated with the given Dataset Version.
+
+    Args:
+        ctx (Context): CLI context
+        id (str): Dataset ID
+        version_id (str): Dataset version ID
+    """
```
[the remark this answers is not in the document]
Comment (iroper-tessella) [reply dafni_cli/commands/download.py:42]: Could you also add a "\f" after the first line to ensure the arguments aren't printed?

File: test/test_utils.py (hunk 1 above)
Comment (jamie-short) [reply test/test_utils.py:788]: So `tmp_path` is a fixture supplied by the pytest module (https://docs.pytest.org/en/stable/tmpdir.html), basically sets up a temporary folder for the test then cleans it up after completion.

File: dafni_cli/utils.py [hunk 3]
```diff
@@ -179,20 +182,23 @@ def output_table(
     return table_str
 
 
-def process_file_size(file_size: Union[int, float]) -> str:
+def process_file_size(file_size: str) -> str:
```
[the remark this answers is not in the document]
Comment (jamie-short) [reply dafni_cli/utils.py:185]: updated to cast to str, float and int as inputs

**Comment by jamie-short** [comment]
Updates made requested by Ian

**Pull request merged**